matrix-sdk-android/src/main/java/im/vector/matrix/android/internal/session/sync/SyncResponseHandler.kt

```diff
@@ -106,13 +110,30 @@ internal class SyncResponseHandler @Inject constructor(private val monarchy: Mon
 
         // Everything else we need to do outside the transaction
         syncResponse.rooms?.also {
-            checkPushRules(it, isInitialSync)
-            userAccountDataSyncHandler.synchronizeWithServerIfNeeded(it.invite)
+            coroutineScope.processRoomSyncResponse(it, isInitialSync)
         }
-        Timber.v("On sync completed")
         cryptoSyncHandler.onSyncCompleted(syncResponse)
     }
 
+    private fun CoroutineScope.processRoomSyncResponse(syncResponse: RoomsSyncResponse, isInitialSync: Boolean) = launch {
+        checkPushRules(syncResponse, isInitialSync)
+        userAccountDataSyncHandler.synchronizeWithServerIfNeeded(syncResponse.invite)
+        val processMode = if (isInitialSync) {
+            RoomEventsProcessor.Mode.INITIAL_SYNC
+        } else {
+            RoomEventsProcessor.Mode.INCREMENTAL_SYNC
+        }
+        syncResponse.join.forEach {
+            val roomId = it.key
+            val roomSync = it.value
+            roomEventsProcessors.process(processMode, roomId, roomSync.timeline?.events ?: emptyList())
+            roomEventsProcessors.process(processMode, roomId, roomSync.accountData?.events ?: emptyList())
+            roomEventsProcessors.process(processMode, roomId, roomSync.ephemeral?.events ?: emptyList())
```

Comment:
ephemeral events are processed?

Reply:
Not at the moment, not sure it's relevant to keep account and ephemeral

Reply:
COuld we have 3 methods? 
processTimelineEvent
processRoomAccountDataEvent
processEphemeralEvents